sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs

```diff
@@ -788,6 +792,20 @@ public enum ShareFileRangeWriteType
         Update = 0,
         Clear = 1,
     }
+    public partial class ShareFileRenameOptions
+    {
+        public ShareFileRenameOptions() { }
+        public string ContentType { get { throw null; } set { } }
```

Comment:
`ShareFileRenameOptions` is used for both File and Directory Rename.  `ContentType` is not settable for a directory, so maybe we should throw an ArguementException in `ShareDirectoryClient.Rename()` if the user attempts to set it?

Reply:
If `ContentType` is not settable for Directories, shouldn't there be two different option bags then? `ShareFileRenameOptions` and `ShareDirectoryRenameOptions` that way the user knows what parameters can be set?

Some customers see the SDK as the definition of what the service accepts (not that I am approving of why they think like this).

Reply:
@amnguye, maybe we could just pull `ContentType` from this type?  I guess Rename Directory and Rename File could also further diverge, so maybe we should make 2 different types.